frontend/src/pages/UserDashboard/UserDashboard.jsx

```diff
@@ -0,0 +1,131 @@
+            console.log(profileData);
+            setProfileData(profile);
+          }
+        } else if (activeTab === 'Home') {
+          if (userId) {  
+            const courses = await fetchNotEnrolledCourses(userId);
+            setCoursesData(courses);
+          }
+        } else if (activeTab === 'My Courses') {
+          if (userId) {            
+            const enrolledCourses = await fetchEnrolledCourses(userId);
+            setEnrolledCoursesData(enrolledCourses);
+          }
+        }
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+
+    fetchData();
+  }, [activeTab, userId]);
+
+  const handleCourseClick = (course) => {
+    const isEnrolled = activeTab === 'My Courses'; 
+    navigate(`/course/${course.courseName}`, { state: { course, isEnrolled } });
+  };
+
+  const renderStars = (rating) => {
+    return Array.from({ length: 5 }, (_, index) => (
+      <span key={index} className={index < rating ? styles.starFilled : styles.starEmpty}>★</span>
+    ));
+  };
+
+  return (
+    <div className={styles.dashboardContainer}>
+      <header className={styles.header}>
+       
+        <button 
+          className={`${styles.tabButton} ${activeTab === 'Home' ? styles.activeTab : ''}`} 
+          onClick={() => setActiveTab('Home')}
+        >
+          Home
+        </button>
+        <button 
+          className={`${styles.tabButton} ${activeTab === 'My Courses' ? styles.activeTab : ''}`} 
+          onClick={() => setActiveTab('My Courses')}
+        >
+          My Courses
+        </button>
+        <button 
+          className={`${styles.tabButton} ${activeTab === 'Profile' ? styles.activeTab : ''}`} 
+          onClick={() => setActiveTab('Profile')}
+        >
+          Profile
+        </button>
+      </header>
+      <div className={styles.introText}>
+        <h1>Get Started with These Free Courses</h1>
+      </div>
+      <main className={styles.content}>
+      <div className={styles.parentContainer}>
+        {activeTab === 'Profile' && profileData && (
+          <div className={styles.profileContainer}>
+            <h2>Profile Information</h2>
+            <div><img src={profileData.profilePictureUrl} alt="Profile" className={styles.profileImage}/></div>
+            <div className={styles.profileDetails}>
+              <p><strong>Name:</strong> {profileData.name}</p>
+              <p><strong>Email:</strong> {profileData.email}</p>
+              {/* <p><strong>Joined:</strong> {new Date(profileData.createdAt).toLocaleDateString()}</p> */}
+              {/* Add more profile details as needed */}
+            </div>
+          </div>
+        )}
+        {activeTab === 'Home' && (
+          <div className={styles.coursesContainer}>
+            {coursesData.map(course => (
+              <div key={course.courseID} className={styles.courseCard} onClick={() => handleCourseClick(course)}>
```

Comment:
This should be a separate component - a `CourseThumbnail` or something like that.